src/components/common/GoodsItem/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import Image from '@components/common/Image';
+import Ranking from './Ranking';
+
+const IMAGE_SIZE_RANKING = 160;
+const IMAGE_SIZE_GOODS_ITEM = 250;
+const IMAGE_RADIUS = 4;
+
+export interface GoodsItemProps {
+  imageSrc: string;
+  subtitle: string;
+  title: string;
+  amount: number;
+  rankingIndex?: number;
+}
+
+export default function GoodsItem({ imageSrc, subtitle, title, amount, rankingIndex, ...rest }: GoodsItemProps) {
+  const imageSize = rankingIndex ? IMAGE_SIZE_RANKING : IMAGE_SIZE_GOODS_ITEM;
+
+  const renderRanking = () => rankingIndex && <Ranking {...{ rankingIndex }} />;
+
+  return (
+    <StyledGoodsItem rankingIndex={rankingIndex} {...rest}>
+      {renderRanking()}
```

Comment:
```suggestion
      {rankingIndex && <Ranking {...{ rankingIndex }} />}
```

위처럼 inline 할 수 있을텐데 컴포넌트 내부의 함수로 추출해야하는 이유는 어떤걸까요?